### ISSUE TITLE
index.d.ts ts(2714) - The expression of an export assignment must be an identifier or qualified name in an ambient context.

### ISSUE THREAD

**Issue by laurynas-wn** [ferdikoomen/openapi-typescript-codegen#957]
I've been using version 0.12.5 for a while and wanted to update. It seems  you changed the way types are exported and I'm getting an error  `ts(2714) - The expression of an export assignment must be an identifier or qualified name in an ambient context.`  on

```
export default {
    HttpClient,
    Indent,
    generate,
};
```
in types/index.d.ts.

Is my ts setup incorrect? You seem to have changed it at 0.13, and quite a bit of peope are using 0.13 or later.

**Comment by ferdikoomen** [comment]
@laurynas-wn How are you using the openapi generator? As command inside `package.json` or using it as a node api? And what version of Node.js are you using?

**Comment by laurynas-wn** [comment]
I'm using it during runtime, so as a node api. 

Node version: 16.10
Typescript: 4.4.4

**Comment by ferdikoomen** [comment]
Just to be sure, you are doing something like this:

```
const OpenAPI = require('openapi-typescript-codegen');

OpenAPI.generate({
    input: './spec.json',
    output: './generated',
});
```
Can you post your `tsconfig.json` file here, then ill have a look!

**Comment by laurynas-wn** [comment]
Yup, thanks a lot.
```
  OpenAPI.generate({
      input: './spec-json',
      output: '../generated',
      httpClient: OpenAPI.HttpClient.AXIOS,
    })
```

```
{
  "compilerOptions": {
    "module": "commonjs",
    "declaration": true,
    "removeComments": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "allowSyntheticDefaultImports": true,
    "target": "es2017",
    "sourceMap": true,
    "incremental": true,
    "resolveJsonModule": true,
    "strict": true,
    "outDir": "./dist",
    "baseUrl": "./",
    "paths": {
      "src/*": ["src/*"],
      "test/*": ["test/*"]
    }
  },
  "exclude": ["upload", "src/reservation/"],
  "include": ["src"]
}

```

**Comment by ferdikoomen** [comment]
@laurynas-wn I created a minimal example:
https://filebin.net/eqffzc754t6u8pik
And it seems fine, can you provide a minimal example that has the issue?

**Comment by laurynas-wn** [comment]
@ferdikoomen Your example already has the error, it just doesn't throw it, because you're not compiling ts to js. 
I've updated your example.
https://filebin.net/n3mmhy8oopjjc2st

**Comment by ferdikoomen** [comment]
@laurynas-wn Cool thanks, i thought it was happening in the generation step, thanks for updating the example

**Comment by ferdikoomen** [comment]
@laurynas-wn I pushed a fix to the master branch, will upload a new version soon. I updated the export to:

```
declare type OpenAPI = {
    HttpClient: HttpClient;
    Indent: Indent;
    generate: typeof generate;
};

export default OpenAPI;
```

Now everything is happy indeed. This use-case is a bit new, normally people run this code (inside your `index.ts` example) using Node.js or ts-node to generate the OpenAPI files, they do not compile this generation code to javascript. But indeed it should support this, so thanks for flagging this
